datafusion/expr/src/built_in_function.rs

```diff
@@ -755,7 +755,10 @@ impl BuiltinScalarFunction {
                     return plan_err!("The to_hex function can only accept integers.");
                 }
             }),
-            BuiltinScalarFunction::ToTimestamp => Ok(Timestamp(Nanosecond, None)),
+            BuiltinScalarFunction::ToTimestamp => Ok(match &input_expr_types[0] {
+                dt if dt.is_integer() => Timestamp(Second, None),
```

Comment:
The implementation seems to  handle `Int64`: https://github.com/apache/arrow-datafusion/blob/ae9a446c41dfa515eb454561c160ad9aa26a7117/datafusion/physical-expr/src/functions.rs#L75-L98

I would expect this code to follow the same semantics. 

So perhaps something like (untested):

```suggestion
                &DataType:Int64 => Timestamp(Second, None),
```


(it looks like UTf8 is already covered by _ as passing a UTF8 results in a timestamp nanosecond)

What would you think about changing the implementation to always return Timestamp(Nanoseconds)? even for an integer argument?

Reply:
Testing integers.
We cannot return always nanoseconds, in that case we limit the output as cannot represent timestamps far in the past or in the future. example https://github.com/apache/arrow-datafusion/issues/7802 and also its incompatible with PG https://github.com/apache/arrow-datafusion/issues/2979